pkg/operator/controller/controller.go

```diff
@@ -393,21 +406,45 @@ func (r *ReconcileCDI) checkReady(logger logr.Logger, cr *cdiv1alpha1.CDI) (bool
 			one := int32(1)
 			desiredReplicas = &one
 		}
+		logger.Info("Desired", "replicas", desiredReplicas)
 
 		if *desiredReplicas != deployment.Status.Replicas ||
 			deployment.Status.Replicas != deployment.Status.ReadyReplicas {
-			readyCond = conditionNotReady
+			logger.Info("Degraded set to true", "replicas", deployment.Status.Replicas)
+			logger.Info("ReadyReplicas", "replicas", deployment.Status.ReadyReplicas)
+			degraded = true
 		}
-
 	}
 
-	logger.Info("CDI Ready check", "Status", readyCond.Status)
+	logger.Info("CDI degraded check ", "Degraded", degraded)
 
-	if err = r.conditionUpdate(readyCond, cr); err != nil {
-		return false, err
+	if degraded {
```

Comment:
can this block be cleaned up somehow?  try to minimize the calls to `r.crUpdate`

Reply:
Just making sure I understand what you are asking here. You want me to make sure I try not to call r.crUpdate if no status/conditions actually changed?

Reply:
No, there are three places in that block that do this:
```golang
if err = r.crUpdate(<some phase value>, cr); err != nil {
	return degraded, err
}
```
Can that be reduced?  If not, can the code be extracted to a helper function?